Refactor and deduplicate stream notification settings.

YJDave opened the pull request:
* Rename notification property `enable_stream_sounds` to
`enable_stream_audible_notifications` to match with other
notification property patterns.

Fixes part of #12304

Tested manually with stream notification settings.

YJDave commented:
While I am working on further refactoring and deduplication, pushed commits are ready for review.

timabbott commented:
Awesome!  I merged the first two commits.  For the third commit, I think the handling of `end_content` is incorrect; I think b553507412352b516f55f4aae977ec23a237b206 neglected to fully remove `propagate_stream_notifications_change` from the codebase, and you should just put a prep commit in that fully removes it, and then not need to reference that in the `settings_checkbox` templates.  Might be able to also remove the `end_content` concept entirely in an additional prep commit, since I don't think it's used for anything else.

I think it might also be worth splitting that first commit into a first commit that does the reordering of things like `pm_mention_notification_settings` to match the template, and a second that does the deduplication, just because it'd be nice for me to able to just verify the ordering and merge that first piece.  

Thanks @YJDave!

timabbott commented:
Also @pragatiagrawal31 FYI, you'll want to avoid merge conflicting with this.

YJDave commented:
@timabbott PR is updated with removal of duplicates in various files.

timabbott commented:
I merged the first few commits, after fixing the spelling of "notification" (was notificaion).

timabbott commented on static/templates/settings/settings_checkbox.handlebars:
```diff
@@ -1,3 +1,4 @@
+{{#if_a_then_b has_render_condition render_only}}
```
I'm not sure why this hunk is in this commit; it seems like it's misplaced.

timabbott commented:
I pushed the remaining commits back to this PR.  I think you have some more `stream_notificaion_settings` typos to fix too.

timabbott commented:
I also don't totally understand why why we need the `if_a_then_b` helper; can you explain that?

YJDave commented:
@timabbott Thanks for fixes and review! I fixed the typos, sorry didn't noticed first time.
I removed `if_a_then_b` helper, cause I was writing an explanation and get idea that already existing helper `is_false` will work, we don't need another. 
The use case is that if `render_only` condition is set to false then we should not render setting. But if it is not set, undefined, then we should render setting, cause there is no condition. So `is_false` flag work.

pragatiagrawal31 commented:
I had made a basic prototype for #12182 , but seems like @YJDave changes will require me to rework those changes, so I'm putting #12182 on hold and will resume my work after this PR gets merged.

YJDave commented:
Ok, I have updated this PR to include solution of #12182

shubhamdhama commented:
> Ok, I have updated this PR to include solution of #12182

I don't think it solves #12182(judging from commits), but before you start I want to mention @pragatiagrawal31 has already started working on this, mentioning this to prevent any conflict.

YJDave commented:
Oh, I mean #12208 not #12182

YJDave commented:
Oh, I mean #12208 not #12182

timabbott commented:
This is awesome, merged, thanks @YJDave!  

Are there more things like this to clean up?  Once we're done with this big refactoring project, probably @rishig and/or I should do a QA run through "display settings" and "stream settings" to make sure we don't have any regressions, but I'm not sure now's the time.
